### PR TITLE
Machine api mirror

### PR DESCRIPTION
**Pull request by enxebre**
This mirrors the upstream API into `machine.openshift.io` and update the controllers to consume it
This give us the autonomy we need today but does not preclude our ability to adopt the upstream project at a later time

### CODE REVIEW

**Comment by enxebre** [comment]
ping @frobware @ingvagabund @bison

File: pkg/apis/machine/v1beta1/v1alpha1_suite_test.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
+/*
```
Comment (ingvagabund) [inline pkg/apis/machine/v1beta1/v1alpha1_suite_test.go:1]: Just a nit.  `s/v1alpha1_suite_test.go/v1beta1_suite_test.go`. All just keep it as it is.

**Pull request merged**